src/main/jbake/content/blog/2017/09/agent-to-services-over-ssl.adoc

```diff
@@ -0,0 +1,146 @@
+
+```
+docker run --name hawkular-cassandra -e CASSANDRA_START_RPC=true -d cassandra:3.0.9
+```
+
+== Starting Hawkular Services on SSL
+
+Now that there is a cassandra ready, Hawkular Services can be started,
+it will be linked to the cassandra instance (hawkular-cassandra)
+
+```
+docker run --name hawkular-services --link=hawkular-cassandra -e CASSANDRA_NODES=hawkular-cassandra -e HAWKULAR_HOSTNAME=hawkular-services -e HAWKULAR_USE_SSL=true -p 8443:8443 -v `pwd`/hawkular-services-private.key:/client-secrets/hawkular-services-private.key -v `pwd`/hawkular-services-public.pem:/client-secrets/hawkular-services-public.pem hawkular/hawkular-services
+```
+
+NOTE: To avoid guessing the host, we explicitly set it to hawkular-services
+using HAWKULAR_HOSTNAME environmental variable.
+
+WARNING: If you don't specify any certificate, Hawkular services will create one,
+but we won't be able to connect agents unless we export the certificate.
+
+== Starting a Hawkular Agent
+
+By now there should be a Hawkular Services listening on default secure port 8443, if any agent wants to connect,
+it will need to know and trust its certificate.
+If you are self-signing your certificate, you will need to pass Hawkular Service's certificate when starting the agent.
+
+```
+docker run --name hawkular-agent-01 --link=hawkular-services -e HAWKULAR_SERVER_PROTOCOL=https -e HAWKULAR_SERVER_ADDR=hawkular-services -e HAWKULAR_SERVER_PORT=8443 -v `pwd`/hawkular-services-public.pem:/client-secrets/hawkular-services-public.pem wildfly-hawkular-javaagent
+```
+
+NOTE: Host must match with the one specified in the certificate, thus setting HAWKULAR_SERVER_ADDR to hawkular-services
+is required. Update as needed if using other value.
+
+== Testing the setup
+
+Any Hawkular client that supports connecting using SSL can be used. ManageIQ and HawkFX will be show below.
+
+NOTE: If using self-signed certificates, the client machine needs to trust Hawkular Services certificate or the client
+should support to specify which certificate to trust.
+See link:http://www.hawkular.org/blog/2016/09/14/consuming-hawkular-api-over-ssl.html[here]
+for more info.
+
+=== ManageIQ
+
+It can be quickly tested using the dockerized ManageIQ as follow:
+
+```
+docker run --link=hawkular-services --privileged -d -p 8444:443 manageiq/manageiq:fine-3
+```
+
+Once ManageIQ has started, add a Middleware Provider, there is three secured ways to do that:
+
+. SSL  -- Requires that the client box trusts Hawkular Services certificate.
+. SSL trusting custom CA -- Requires to provide the certificate to trust.
+. SSL without validation -- No validation is performed.
+
+We will focus on (2) and (3) as (1) will require to trust the certificates on the machine itself.
+
+==== SSL trusting custom CA
+
+We need to select `SSL trusting custom CA` in `Security Protocol`
+and copy the certificate from the contents of `hawkular-services-public.pem`.
```

Comment:
Can't get it work :-(
It says "Unable to verify credentials"

Reply:
PS: probably linked to my first comment, it seems it didn't take my PEM files.

Reply:
Yes, related, I'll verify.